projects/cdk/directives/auto-focus/handlers/abstract.handler.ts

```diff
@@ -1,16 +1,19 @@
 import type {ElementRef} from '@angular/core';
 
-import type {TuiAutofocusHandler} from '../autofocus.options';
+import type {TuiAutofocusHandler, TuiAutofocusOptions} from '../autofocus.options';
 
 export abstract class AbstractTuiAutofocusHandler implements TuiAutofocusHandler {
-    constructor(protected readonly el: ElementRef<HTMLElement>) {}
+    constructor(
+        protected readonly el: ElementRef<HTMLElement>,
+        protected readonly options: TuiAutofocusOptions,
+    ) {}
 
     public abstract setFocus(): void;
 
     protected get element(): HTMLElement {
         // TODO: Remove when legacy controls are dropped
         const el = this.el.nativeElement.tagName.includes('-')
-            ? this.el.nativeElement.querySelector<HTMLElement>('input,textarea')
+            ? this.el.nativeElement.querySelector<HTMLElement>(this.options.query)
```

Comment:
Even if you specify custom query here, `isTextFieldElement()` check would prevent it from working.

Reply:
fixed